src/libertem/udf/base.py

```diff
@@ -456,6 +471,36 @@ def set_meta(self, meta):
     def set_slice(self, slice_):
         self.meta.slice = slice_
 
+    def set_coordinates(self, slice_, ds_shape, roi):
```

Comment:
Conceptually, I think it might make sense to have the coordinates available not only via the `UDFMeta`, but also for every consumer of the `DataTile` stream (i.e. everyone who calls `get_tiles`), if possible. Would it maybe make sense to calculate the coords in a method/property of `DataTile`? And then also provide it in `UDFMeta`, similar to how `set_slice` works (just a simple setter).

What do you think?

/cc @uellue

Reply:
Perhaps it would be better to offer this as a function that a UDF can call with the required information? Most UDFs actually don't need it, so it would save us some processing time in a relatively "hot" loop.

Reply:
So it should be a method/property of `DataTile` which a UDF should be able to call to get the coordinates, right?

Reply:
Yes, just not necessarily of the `DataTile`. The current `set_coordinates()` method could actually become a `get_coordinates()` method of `UDFMeta` since all current parameters are actually properties of `UDFMeta`, right?

As an additional remark, this method should probably also get a benchmark since it might potentially be called very often. That way we make sure it is within reasonable limits, and it helps to catch any regressions. The benchmarks are based on `pytest` like the unit tests and are in the `benchmarks` folder, in case you haven't seen them yet. :-)

Reply:
...as @sk1p already suggested! 🙈

Reply:
> So it should be a method/property of `DataTile` which a UDF should be able to call to get the coordinates, right?

> Yes, just not necessarily of the `DataTile`.

Maybe this index transform code could be in its own class, which can then be used and tested independently of the UDF interface, but still be made available in `UDFMeta`.

(I was thinking that maybe we could actually move `UDFMeta.slice` and the new coordinate stuff to `DataTile`, and make `DataTile` part of the external interface. For by-frame processing, we could slice into the `DataTile`, yielding a new `DataTile` that is a view into the tile, including coordinates. But that doesn't have to happen in this PR, if ever)

Reply:
I've separated the coordinates generation code from the UDF interface. So the UDFs which need the coordinates can `self.set_coordinates(self.meta.slice, self.meta.dataset_shape)` and the coords would be available in the `UDFMeta`. Other functions outside the UDF interface can use `libertem.io.dataset.base import get_coordinates` directly. The tests should make this more clear. Please take a look, thanks!